pandas/io/formats/format.py

```diff
@@ -1198,6 +899,209 @@ def _get_column_name_list(self) -> List[str]:
+
+        html_formatter = Klass(
+            self.fmt,
+            classes=classes,
+            border=border,
+            table_id=table_id,
+            render_links=render_links,
+        )
+        string = html_formatter.to_string()
+        return self._get_result(string, buf=buf, encoding=encoding)
+
+    def to_string(
+        self,
+        buf: Optional[FilePathOrBuffer[str]] = None,
+        encoding: Optional[str] = None,
+        line_width: Optional[int] = None,
+    ) -> Optional[str]:
+        """
+        Render a DataFrame to a console-friendly tabular output.
+
+        Parameters
+        ----------
+        buf : str, Path or StringIO-like, optional, default None
+            Buffer to write to. If None, the output is returned as a string.
+        encoding: str, default “utf-8”
+            Set character encoding.
+        line_width : int, optional
+            Width to wrap a line in characters.
+        """
+        from pandas.io.formats.string import StringFormatter
+
+        string_formatter = StringFormatter(self.fmt, line_width=line_width)
+        string = string_formatter.to_string()
+        return self._get_result(string, buf=buf, encoding=encoding)
+
+    def to_csv(
+        self,
+        path_or_buf: Optional[FilePathOrBuffer[str]] = None,
+        encoding: Optional[str] = None,
+        sep: str = ",",
+        columns: Optional[Sequence[Label]] = None,
+        index_label: Optional[IndexLabel] = None,
+        mode: str = "w",
+        compression: CompressionOptions = "infer",
+        quoting: Optional[int] = None,
+        quotechar: str = '"',
+        line_terminator: Optional[str] = None,
+        chunksize: Optional[int] = None,
+        date_format: Optional[str] = None,
+        doublequote: bool = True,
+        escapechar: Optional[str] = None,
+        errors: str = "strict",
+        storage_options: StorageOptions = None,
+    ) -> Optional[str]:
+        """
+        Render dataframe as comma-separated file.
+        """
+        from pandas.io.formats.csvs import CSVFormatter
+
+        csv_formatter = CSVFormatter(
+            path_or_buf=path_or_buf,
+            line_terminator=line_terminator,
+            sep=sep,
+            encoding=encoding,
+            errors=errors,
+            compression=compression,
+            quoting=quoting,
+            cols=columns,
+            index_label=index_label,
+            mode=mode,
+            chunksize=chunksize,
+            quotechar=quotechar,
+            date_format=date_format,
+            doublequote=doublequote,
+            escapechar=escapechar,
+            storage_options=storage_options,
+            formatter=self.fmt,
+        )
+        csv_formatter.save()
+
+        if path_or_buf is None:
+            assert isinstance(csv_formatter.path_or_buf, StringIO)
+            return csv_formatter.path_or_buf.getvalue()
+
+        return None
+
+    def _get_result(
```

Comment:
these can probably be non-private. This entire class is private, so that if these are called by other modules (from pandas) then we want to make the public. If something is entirely private to this class then this is ok.

Reply:
It turns out that ``_get_result`` and ``_get_buffer`` should not belong to the class: they do not use the object state. So I decided to move the methods to the module level.